PySDM/dynamics/relaxed_velocity.py

```diff
@@ -4,26 +4,57 @@
 towards the terminal velocity
 """
 
-from math import exp
+import numpy as np
 
 from PySDM.attributes.impl.attribute import Attribute
 from PySDM.particulator import Particulator
 
 
 class RelaxedVelocity:  # pylint: disable=too-many-instance-attributes
     """
-    A dynamic which updates the fall momentum according to a relaxation timescale `tau`
+    A dynamic which updates the fall momentum according to a relaxation timescale
+    proportional to the sqrt of the droplet radius.
+
+    Should be added first in order to ensure the correct attributes are selected.
     """
 
-    def __init__(self, tau):
-        self.tau: float = tau
+    def __init__(self, c: float = 8, constant: bool = False):
+        """
+        Parameters:
+            - constant: use a constant relaxation timescale for all droplets
+            - c: relaxation timescale if `constant`, otherwise the proportionality constant
+        """
+        # the default value of c is a very rough estimate
+        self.c: float = c
+        self.constant = constant
 
         self.particulator = None
         self.fall_momentum_attr = None
         self.terminal_vel_attr = None
         self.volume_attr = None
+        self.radius_attr = None
         self.rho_w = None  # TODO #798 - we plan to use masses instead of volumes soon
-        self.tmp_data = None
+
+        self.tmp_momentum_diff = None
+        self.tmp_tau = None
+        self.tmp_scale = None
+
+    def calculate_tau(self, output, radius_storage):
+        """
+        Calculates the relaxation timescale.
+        """
+        # TODO: this should be done with backend storage functions if possible
+        if not self.constant:
+            output[:] = self.c * np.sqrt(radius_storage.to_ndarray())
```

Comment:
perhaps, we could add a `SquareRootOfRadius` attribute and use it here

Reply:
What would be the benefit of adding another attribute versus computing it directly here?

Reply:
Here, Numpy likely allocates several new sd-long arrays in every timestep: one to store the result of `to_ndarray`, one to store the result of `sqrt` and one to store the result of multiplication before assigning it to `output` (?).

Within attribute code, we strive not to have any allocations at all (i.e., only once per whole simulation for a given attribute).

Additionally, in case a square root of radius would be needed elsewhere in the code, the computation would be reused.

Reply:
Sounds good, I added the `SquareRootOfRadius` attribute (and a unit test). I still have a note saying it should use backend functions if possible, should I make this an issue so the TODO has a corresponding issue?

Reply:
see https://github.com/open-atmos/PySDM/pull/1114/commits/3d25785843ba27f22e9ca60c19d9629689578d96